### PR TITLE
[bitnami/redis] Add VIB tests

### PR DESCRIPTION
**Pull request by mdhont**
Signed-off-by: mdhont [michield@vmware.com](mailto:michield@vmware.com)

### Description of the change

The main objective of this PR is to publish our Bitnami MySQL container using VMware Image Builder. In order to do that, several changes are included:

Increasing the existing test coverage of the asset by adding Goss tests.
Update verify and publish VIB pipeline's definitions.

### Benefits
Ensuring higher quality of the container catalog.
Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

Automated tests could introduce additional flakiness to the CI/CD.

### CODE REVIEW

File: .vib/redis/goss/vars.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  - gosu
+  - redis-benchmark
+  - redis-cli
+  - redis-server
+  - wait-for-port
+directories:
+  - mode: "0775"
+    paths:
+      - /bitnami/redis/data
+      - /opt/bitnami/redis/etc
```
Comment (FraPazGal) [inline .vib/redis/goss/vars.yaml:11]: I think there are some inconsistencies here compared to the permissions modifications and folder creations done at the postunpack stage. Could you please review this?

Comment (mdhont) [reply .vib/redis/goss/vars.yaml]: I've added the paths of the folders that are created in the postunpack stage.  Although I'm not sure what you mean with inconsistencies in permission modifications.

File: .vib/redis/goss/redis.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+file:
+  /opt/bitnami/redis/etc/redis.conf:
+    exists: true
+    mode: "0664"
+    contains:
+      - port {{ .Vars.port }}
```
Comment (joancafom) [inline .vib/redis/goss/redis.yaml:6]: Do not use a variable here. You can replace it by the port instead, although I would personally drop this line as there is no real need of checking every modification in the file

File: .vib/redis/goss/redis.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      - dir /bitnami/redis/data
+      - pidfile /opt/bitnami/redis/tmp/redis.pid
+      - daemonize yes
+command:
+  check-redis-server:
+    exec: timeout 5s redis-server || true
```
Comment (joancafom) [inline .vib/redis/goss/redis.yaml:12]: Why are you using `|| true` here?

File: .vib/redis/goss/vars.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+      - /bitnami
+      - /bitnami/redis
+      - /bitnami/redis/data
+      - /opt/bitnami/redis
+      - /opt/bitnami/redis/etc
+root_dir: /opt/bitnami
+version:
+  bin_name: redis-server
+  flag: --version
+port: 6379
```
Comment (joancafom) [inline .vib/redis/goss/vars.yaml:19]: Remove this line

**Review by joancafom — CHANGES_REQUESTED** [review]
Thanks for your changes! I left some comments as a review

**Review by joancafom — CHANGES_REQUESTED** [review]
Thanks for your changes! I left some comments as a review

File: .vib/redis/goss/redis.yaml (hunk 3 above)
Comment (mdhont) [reply .vib/redis/goss/redis.yaml:12]: Timeout does not exit with status 0. Using || true, seems cleaner.

File: .vib/redis/goss/redis.yaml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      - /dir.*/bitnami/redis/data/
+      - /pidfile.*/opt/bitnami/redis/tmp/redis.pid/
+      - daemonize yes
+command:
+  check-redis-server:
+    exec: timeout 5s redis-server || true
+    exit-status: 0
+    stdout:
+      - Ready to accept connections
+  check-redis-server-ssl:
```
Comment (FraPazGal) [inline .vib/redis/goss/redis.yaml:16]: I'm not sure about these 2 ssl checks, there is no related compilation or postunpack logic that requires it right?

File: .vib/redis/goss/redis.yaml (hunk 3 above)
Comment (FraPazGal) [reply .vib/redis/goss/redis.yaml:12]: If we are configuring Redis to be used as a daemon, couldn't we initialize it so using the `redis.conf` file?

File: .vib/redis/goss/redis.yaml (hunk 5 above)
Comment (mdhont) [reply .vib/redis/goss/redis.yaml:16]: I've added those based on review comments:

https://github.com/bitnami/containers/pull/28158#discussion_r1149286198

File: .vib/redis/goss/redis.yaml (hunk 3 above)
Comment (mdhont) [reply .vib/redis/goss/redis.yaml]: I've changed it and it's better, as it exits properly, and then we can check if the process is running.

**Review by joancafom — APPROVED** [review]
LGTM!

**Review by FraPazGal — APPROVED** [review]

**Pull request merged**